### PR TITLE
allow globals to be specified in jshintrc

### PR DESCRIPTION
**Pull request by eblahm**
also see:
  https://github.com/jshint/jshint/blob/a79d5dbfd2442dd298874b5e006cccaec16f5fc9/src/cli.js#L290

### CODE REVIEW

File: src/main/resources/jshint-shell.js [hunk 1]
```diff
@@ -28,6 +28,13 @@
     var OPTIONS_ARG = 4;
 
     var options = JSON.parse(stripJsonComments(args[OPTIONS_ARG]));
+    var globals;
+
+    if (options.globals) {
+        globals = options.globals;
+        delete options.globals;
```
Comment (huntc) [inline src/main/resources/jshint-shell.js]: Why delete here?

Comment (eblahm) [inline src/main/resources/jshint-shell.js]: I was just following the pattern of the cli.js in the jshint repo. 
https://github.com/jshint/jshint/blob/a79d5dbfd2442dd298874b5e006cccaec16f5fc9/src/cli.js#L282
It doesn't make a functional difference right now.  However, if in the future... JSHint module decides to iterate over options somewhere later on, not deleting `options.globals` could cause an error.

Comment (huntc) [inline src/main/resources/jshint-shell.js]: Thanks for the explanation. However assign one object to another will not cause a copy - it is just assigning a reference, correct? So I stll don't understand the deletion... sorry if this is something obvious.

Comment (eblahm) [inline src/main/resources/jshint-shell.js]: I believe its just assigning a reference to `options.globals`.... but somehow JavaScript knows how to manage that memory even if `options` no longer has a `globals` property.   I'm not sure that my explanation is accurate.  

I think what matters here is that `options` no longer has a `globals` property
```
$ node
> var options = {undef: true, globals: {"$": true}};
undefined
> var globals = options.globals;
undefined
> delete options.globals;
true
> globals
{ '$': true }
> options.globals
undefined
> options
{ undef: true }
```

whats funny is that `globals` is not a copy. as you say.
```
> var options = {undef: true, globals: {"$": true}};
undefined
> var globals = options.globals;
undefined
> options.globals.define = true
true
> globals
{ '$': true, define: true }
```

Comment (eblahm) [inline src/main/resources/jshint-shell.js]: Apparently, delete just removes the property the object.  It doesn't have anything to do with memory.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

**Comment by huntc** [comment]
Hi @eblahm,

Thank you for your contribution.  Would you mind signing our CLA?

http://www.typesafe.com/contribute/cla

Thanks!

**Comment by eblahm** [comment]
done

**Comment by huntc** [comment]
Thanks for the Moz link. JS is such a peculiar language. :-)

**Pull request merged**